src/walletConnect/request.ts

```diff
@@ -109,12 +187,35 @@ export function* handleRequest({ method, params }: { method: string; params: any
       )
       return receipt.transactionHash
     }
-    case SupportedActions.personal_sign:
-      return (yield* call([wallet, 'signPersonalMessage'], account, params[0])) as string
-    case SupportedActions.eth_sign:
-      const web3: Web3 = yield* call(getWeb3)
-      return (yield* call(web3.eth.sign.bind(web3), params[1], account)) as string
+    case SupportedActions.eth_signTypedData_v4:
+    case SupportedActions.eth_signTypedData:
+      return (yield* call([wallet, 'signTypedData'], JSON.parse(params[1]))) as string
+    case SupportedActions.personal_sign: {
+      const data = { message: { raw: params[0] } } as SignMessageParameters
+      return (yield* call([wallet, 'signMessage'], data)) as string
+    }
+    case SupportedActions.eth_sign: {
+      const data = { message: { raw: params[1] } } as SignMessageParameters
+      return (yield* call([wallet, 'signMessage'], data)) as string
+    }
```

Comment:
Could we also wrap these in `useViem`?
I'd prefer it controlled all WC actions, this way we keep the old implementation in case something goes wrong.